smart_open/s3.py

```diff
@@ -1171,33 +1166,31 @@ def close(self):
                 'the bucket %r does not exist, or is forbidden for access' % self._bucket) from e
 
         logger.debug("%s: direct upload finished", self)
-        self._buf = None
+        self._buf.close()
```

Comment:
such that calling `seek`, `seekable` etc at a later point in time will properly raise `ValueError: I/O operation on closed file`